File: codewars/6kyu/break-camel-case/dduck-break-camel-case.py
# https://www.codewars.com/kata/5208f99aee097e6552000148/

from unittest import TestCase

import re

solution = lambda s: re.sub(pattern="([A-Z])", repl=" \\1", string=s)

TestCase().assertEqual(solution("helloWorld"), "hello World")
TestCase().assertEqual(solution("camelCase"), "camel Case")
TestCase().assertEqual(solution("breakCamelCase"), "break Camel Case")

File: codewars/6kyu/break-camel-case/dduck-break-camel-case.js
// https://www.codewars.com/kata/5208f99aee097e6552000148/

// complete the function
const solution = (string) => {
  const addSpace = (match) => {
    return ` ${match}`;
  };
  return string.replace(/[A-Z]/g, addSpace);
};

console.log(solution('camelCasing'), 'camel Casing', 'Unexpected result');
console.log(solution('camelCasingTest'), 'camel Casing Test', 'Unexpected result');

// best practice in codewars
// it's capture and return captured value with space
//
// function solution(string) {
//   return string.replace(/([A-Z])/g, ' $1');
// }
